### ISSUE TITLE
Appender.CreateDir doesn't work when Appender.UseLockFile=true

### ISSUE THREAD

**Issue by Jormack** [log4cplus/log4cplus#86]
While using Lock files I get this error when the directory "logs" doesn't exist "log4cplus:ERROR could not open or create file /logs/some_name.log.lock"

Both `CreateDirs` and `UseLockFile` work correctly independently, but together the directory is not created prior to attempting to write to the lock file.

Example properties file:

~~~~
log4cplus.appender.Name=log4cplus::RollingFileAppender
log4cplus.appender.Name.CreateDirs=true
log4cplus.appender.Name.File=./logs/some_name.log
log4cplus.appender.Name.UseLockFile=true
log4cplus.appender.Name.MaxFileSize=100MB
log4cplus.appender.Name.MaxBackupIndex=10
~~~~

**Comment by wilx** [comment]
That sounds like a genuine bug. I will take a look at it.

**Comment by wilx** [comment]
This should be fixed now in master and 1.2.x.
